client/guidestours/positioning.js

```diff
@@ -0,0 +1,61 @@
+const BULLSEYE_RADIUS = 6;
+const DIALOG_WIDTH = 410;
+const DIALOG_PADDING = 10;
+const MASTERBAR_HEIGHT = 48;
+
+const middle = ( a, b ) => Math.abs( b - a ) / 2;
+
+const wouldBeOffscreen = pos =>
+	( pos + DIALOG_PADDING + DIALOG_WIDTH ) > document.documentElement.clientWidth
+
+const ensureOnScreen = pos =>
+	Math.abs( pos - DIALOG_PADDING - DIALOG_WIDTH )
+
+const dialogPositioners = {
+	below: ( { left, bottom } ) => ( {
+		x: wouldBeOffscreen( left )
+			? ensureOnScreen( document.documentElement.clientWidth )
+			: ( left + DIALOG_PADDING ),
+		y: bottom + DIALOG_PADDING
+	} ),
+	beside: ( { left, right, top } ) => ( {
+		x: wouldBeOffscreen( right )
+			? ( ensureOnScreen( left ) )
+			: ( right + DIALOG_PADDING ),
+		y: top + DIALOG_PADDING
+	} ),
+	center: ( { left, right } ) => ( {
+		x: middle( left, right ) - DIALOG_WIDTH / 2,
+		y: MASTERBAR_HEIGHT / 2,
+	} ),
+};
```

Comment:
Some of the functions above use trailing commas in their objects, some don't. Can we pick one? :) (I vote for trailing commas)